Frameworks/UIKit.Xaml/TextField.xaml.cpp

```diff
@@ -0,0 +1,759 @@
+    }
+}
+
+} /* Xaml*/
+} /* UIKit*/
+
+////////////////////////////////////////////////////////////////////////////////////
+// ObjectiveC Interop
+////////////////////////////////////////////////////////////////////////////////////
+
+  // Returns a UIKit::Xaml::TextField as an IInspectable
+UIKIT_XAML_EXPORT void XamlCreateTextField(IInspectable** created) {
+    ComPtr<IInspectable> inspectable = InspectableFromObject(ref new UIKit::Xaml::TextField());
+    *created = inspectable.Detach();
+}
+
+// Returns a UIKit::Xaml::TextField's backing Canvas as an IInspectable
+UIKIT_XAML_EXPORT IInspectable* XamlGetTextFieldSubLayerCanvas(const Microsoft::WRL::ComPtr<IInspectable>& inspectableTextField) {
+    auto textField = safe_cast<UIKit::Xaml::TextField^>(reinterpret_cast<Platform::Object^>(inspectableTextField.Get()));
+    return InspectableFromObject(textField->SublayerCanvas).Detach();
+}
+
+// Retrieves the UIKit::Xaml::TextField's backing TextBox as an IInspectable
+UIKIT_XAML_EXPORT IInspectable* XamlGetTextFieldTextBox(const Microsoft::WRL::ComPtr<IInspectable>& inspectableTextField) {
+    auto textField = safe_cast<UIKit::Xaml::TextField^>(reinterpret_cast<Platform::Object^>(inspectableTextField.Get()));
+    return InspectableFromObject(textField->TextBox).Detach();
+}
+
+// Retrieves the UIKit::Xaml::TextField's backing PasswordBox as an IInspectable
+UIKIT_XAML_EXPORT IInspectable* XamlGetTextFieldPasswordBox(const Microsoft::WRL::ComPtr<IInspectable>& inspectableTextField) {
+    auto textField = safe_cast<UIKit::Xaml::TextField^>(reinterpret_cast<Platform::Object^>(inspectableTextField.Get()));
+    return InspectableFromObject(textField->PasswordBox).Detach();
+}
+
+// Set the UIKit::Xaml::TextField's secureTextEntry property
+UIKIT_XAML_EXPORT void XamlSetTextFieldSecureTextEntryValue(const Microsoft::WRL::ComPtr<IInspectable>& inspectableTextField, bool secureTextEntry) {
+    auto textField = safe_cast<UIKit::Xaml::TextField^>(reinterpret_cast<Platform::Object^>(inspectableTextField.Get()));
+    textField->SecureEntry = secureTextEntry;
+}
+
+// Get the UIKit::Xaml::TextField's secureTextEntry property
+UIKIT_XAML_EXPORT bool XamlGetTextFieldSecureTextEntryValue(const Microsoft::WRL::ComPtr<IInspectable>& inspectableTextField) {
+    auto textField = safe_cast<UIKit::Xaml::TextField^>(reinterpret_cast<Platform::Object^>(inspectableTextField.Get()));
+    return textField->SecureEntry;
+}
+
+// Set the UIKit::Xaml::TextField's Text property
+UIKIT_XAML_EXPORT void XamlSetTextFieldText(const Microsoft::WRL::ComPtr<IInspectable>&  inspectableTextField, const std::wstring& text) {
+    auto textField = safe_cast<UIKit::Xaml::TextField^>(reinterpret_cast<Platform::Object^>(inspectableTextField.Get()));
+    textField->Text = Platform::StringReference(text.c_str());
+    if (!textField->SecureEntry) {
+        textField->TextBox->SelectionStart = textField->Text->Length();
+        textField->TextBox->SelectionLength = 0;
```

Comment:
nit: might want to comment about why you're doing this, and only doing it for TextBoxes